src/s3/explorer/s3Nodes.ts

```diff
@@ -38,6 +39,31 @@ export class S3Node extends AWSTreeNodeBase {
         })
     }
 
+    private async listBucketNodes(): Promise<(S3BucketNode | AWSCommandTreeNode)[]> {
+        try {
+            const response = await this.s3.listBuckets()
+            return response.buckets.map(bucket => new S3BucketNode(bucket, this, this.s3))
+        } catch (err) {
+            if (err.code === 'AccessDenied') {
+                if (this.addedBuckets.length === 0) {
+                    const localizedText = localize('AWS.explorerNode.s3.showBucket', 'Click to show existing bucket')
+                    return [new AWSCommandTreeNode(this, localizedText, 'aws.s3.showBucket', [this])]
+                } 
+
+                return this.addedBuckets
+            } else {
+                throw err
+            }
+        }
+    }
+
+    public async showBucket(bucketName: string): Promise<void> {
+        await this.s3.listFiles({ bucketName })
```

Comment:
Are there any other calls to check validity? Seems kind of weird to make this call if you're not going to cache the result.

Reply:
Yeah I think `headBucket` should work. This was quick PoC which is why I marked it as draft.